src/v/cluster/controller_backend.cc

```diff
@@ -178,6 +178,17 @@ ss::future<> controller_backend::do_bootstrap() {
       });
 }
 
+bool has_node_local_replicas(
```

Comment:
Adding it here because i wanted to comment on the strategy here. 

I don’t see a way to prevent the ping-pong of data. 

Assume this: 

1. Computer 1: User requests a data movement to core-0 from core-1
2. Computer 2: User requests data movement from core-1 to core-0

Depending on ordering, the user might get different results. What is preventing the data ping-pong ? 

Also, we need a high level policy to ignore partition leaders for example.

Reply:
Ah i see the revision_id now!! Thanks for or adding that. 

It would be good to have a document in /docs that explains this movement. (We can later put that on a blog too) 

But it is very nuanced and likely to get lost in this merge commit if we don’t write a document describing it the design for it. 

I’d say we want to just write a quick RFC for this.

Reply:
Things to highlight are 

1. Eventually consistent data on cores
2. Revision ID to protect the data ping-pong

Reply:
I’d also add to this design a place to embed a simple policy structure that says if we want to move only followers and leave the leader partitions as is.